### ISSUE TITLE
(Some?) Multicharacter auto-closing pairs for LaTeX, or let us turn them off?

### ISSUE THREAD

**Issue by fpqc** [Microsoft/vscode#33084]
- VSCode Version: Latest VSCode Insiders build as of 8/24/2017
- OS Version: Windows 10 Build 16273

The [most popular extension for LaTeX](/James-Yu/LaTeX-Workshop) is unable to implement the recommended auto-closing pairings for math and display modes in LaTeX ( `\( ... \)` and `\[ ... \]` respectively (according to the official documentation, direct invocation of math mode with `$ ... $` and `$$ ... $$` is deprecated and has been for years)) as auto-closing pairs due to a [limitation directly in VSCode](/James-Yu/LaTeX-Workshop/issues/235#issuecomment-322962041).

If this isn't possible, could something please be added to at least allow the masking out of some auto-closing pairs, because as they are right now, they get in the way for me, and [I can't disable them without forking the extension](https://github.com/James-Yu/LaTeX-Workshop/issues/248) due to another limitation in VSCode.  The extension maintainer is willing to add this as well, but he said that autoclosing pairs can't be disabled with an option.

**Comment by aeschli** [comment]
You are correct, there's currently no API to change the LanguageConfiguration of a language.
That should go into the LanguageConfiguration API.

**Comment by fpqc** [comment]
@aeschli How about the multi-character auto-closing pair options and the fact that 

`\( ... \)` and `\[ ... \]` 

can't be made to work?  Since you changed the title, should I recreate that issue as a separate one?

**Comment by aeschli** [comment]
We have #583 for multi character autopairs.

Let me add that auto pairs can be implemented programmatically by listening to content changes. See https://github.com/formulahendry/vscode-auto-close-tag/blob/master/src/extension.ts as an example. So if you want a solution now you could go for that in the latex extension.

**Comment by fpqc** [comment]
@aeschli It looks like someone was assigned to that issue then canceled the assignment, and it is over two years old.  Is there any realistic prospect of the vscode team implementing it any time soon, or is this something you're leaving to the community?

**Comment by aeschli** [comment]
It's likely not getting implemented any time soon. What we tackle next depends on the number of requests for it. We count the 👍 on the description.

**Comment by vomout** [comment]
Is there anything happening on this? (I presume this feature is about being able to *disable* the super-annoying auto-closing of brackets without having to fork LaTeX Workshop.) Forcing auto-closing brackets down by throat is a total showstopper.

**Comment by alexandrudima** [comment]
Let's track in #38352 which has more upvotes.

**Comment by fpqc** [comment]
@vomout You can turn off auto closing brackets for LaTeX.  I have to find the setting 
``
"editor.autoClosingBrackets": "never",
``
